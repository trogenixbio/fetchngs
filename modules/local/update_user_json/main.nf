process UPDATE_USER_JSON {
    label 'process_low'

    conda "conda-forge::python=3.9.1"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/python:3.9--1' :
        'biocontainers/python:3.9--1' }"

    input:
    path metadata_json
    path samplesheet_json
    val cloud_prefix
    val pub_internal
    val run_info_id

    output:
    path "metadata_update.json"    , emit: metadata_json_update
    path "samplesheet_update.json" , emit: samplesheet_json_update
    path "versions.yml"            , emit: versions

    script:
    def args = task.ext.args ?: ''
    """
    update_user_json.py $metadata_json $samplesheet_json metadata_update.json samplesheet_update.json $cloud_prefix $pub_internal $run_info_id

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
    END_VERSIONS
    """
}
