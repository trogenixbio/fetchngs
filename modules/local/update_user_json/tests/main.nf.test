nextflow_process {

    name "Test process: UPDATE_USER_JSON"
    script "../main.nf"
    process "UPDATE_USER_JSON"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = file("$baseDir" + "/" + params.modules_testdata_base_path_local + "metadata.json", checkIfExists: true)
                input[1] = file("$baseDir" + "/" + params.modules_testdata_base_path_local + "samplesheet.json", checkIfExists: true)
                input[2] = "az://raw"
                input[3] = "public"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
