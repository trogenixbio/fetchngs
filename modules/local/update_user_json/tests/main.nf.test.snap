{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "metadata_update.json:md5,12d6c17a13edf5bd0eb26d3e31a8286e"
                ],
                "1": [
                    "samplesheet_update.json:md5,dd557241cb05a294941bab12901b681d"
                ],
                "2": [
                    "versions.yml:md5,60478d2155726f3d0a47ab9ae02ac879"
                ],
                "metadata_json_update": [
                    "metadata_update.json:md5,12d6c17a13edf5bd0eb26d3e31a8286e"
                ],
                "samplesheet_json_update": [
                    "samplesheet_update.json:md5,dd557241cb05a294941bab12901b681d"
                ],
                "versions": [
                    "versions.yml:md5,60478d2155726f3d0a47ab9ae02ac879"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-04T15:21:01.899738289"
    }
}